### ISSUE TITLE
Scrolling away from the page clunky behaviour

### ISSUE THREAD

**Issue by AnataBakka** [tachiyomiorg/tachiyomi#6346]
### Steps to reproduce

1. Have reader on left-right mode.
2. Zoom in.
3. Scroll up or down until you reach the side of the page.
4. Remove your finger before starting to scroll again in the same direction of the side (like when switching pages), and then, without removing your finger from the screen, try scrolling in any other direction.

### Expected behavior

You can keep on scrolling as usual.

### Actual behavior

Page gets stuck to the side of the page, unless you remove your finger first and then scroll away from the side.

### Tachiyomi version

0.12.3

### Android version

Android 11

### Device

Realme 6

### Other details

This seems similar to when you switch pages and you are allowed to only scroll left or right, however in this case you're blocked from scrolling to anywhere.
There should be no reason to have a "swipe started from side of page" check in this case (like when switching pages left and right), unless the same code is used for both vertical and horizzontal and it does the above check and then, if you're in the correct mode, enters page switch, otherwise does nothing. That "does nothing" shouldn't be a thing.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.12.3](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.
